Lib/test/test_contextlib.py

```diff
@@ -228,6 +229,8 @@ class A:
         def woohoo(a, b):
             a = weakref.ref(a)
             b = weakref.ref(b)
+            # Allow test to work with a non-refcounted GC
+            gc.collect(); gc.collect(); gc.collect()
```

Comment:
Is there a reason this needs to be done three times?

Reply:
It ensures references are still cleaned up even if they've made it out of the "recently allocated" generation in a GC that tracks generations.

Before we merge, I want to check the rest of the test suite to see if this is still the recommended practice for forcing collection, or if there'sa helper in test.support these days.

Reply:
There is `test.support.gc_collect()`, which does also do it three times.